pkg/controller/add.go

```diff
@@ -54,10 +61,40 @@ func AddToManager(ctx context.Context, mgr manager.Manager) error {
 	return AddToManagerWithOptions(ctx, mgr, &DefaultAddOptions)
 }
 
+// watchInfrastructure watches for Infrastructure changes and triggers the Extension reconciliation.
+func watchInfrastructure(mgr manager.Manager) extensionscontroller.WatchBuilder {
```

Comment:
Please add unit tests for predicate and mapper. You can get some inspiration from the `add_test.go` files in g/g.